diff-hl.el

```diff
@@ -424,56 +425,57 @@ in the source file, or the last line of the hunk above it."
 (defun diff-hl-revert-hunk ()
   "Revert the diff hunk with changes at or above the point."
   (interactive)
-  (save-restriction
```

Comment:
One way to avoid this yet-another shift of all code to the left is to create a function called `diff-hl-revert-hunk-1` and move the existing implementation into it. That might seems awkward, but the diff will be much shorter.